extensions/librdkafka/PublishKafka.cpp

```diff
@@ -289,122 +356,309 @@ bool PublishKafka::configureNewConnection(const std::shared_ptr<KafkaConnection>
     }
-    conn = lease->getConn();
-
-    if (!conn->initialized()) {
-      logger_->log_trace("Connection not initialized to %s, %s, %s", client_id, brokers, topic);
-      if (!configureNewConnection(conn, context, flowFile)) {
-        logger_->log_error("Could not configure Kafka Connection");
-        session->transfer(flowFile, Failure);
-        return;
-      }
-    }
+  }
 
-    if (!conn->hasTopic(topic)) {
-      auto topic_conf_ = rd_kafka_topic_conf_new();
-      auto topic_reference = rd_kafka_topic_new(conn->getConnection(), topic.c_str(), topic_conf_);
-      rd_kafka_conf_res_t result;
-      std::string value;
-      char errstr[512];
-      int64_t valInt;
-      std::string valueConf;
-
-      if (context->getProperty(DeliveryGuarantee, value, flowFile) && !value.empty()) {
-        rd_kafka_topic_conf_set(topic_conf_, "request.required.acks", value.c_str(), errstr, sizeof(errstr));
-        logger_->log_debug("PublishKafka: request.required.acks [%s]", value);
-        if (result != RD_KAFKA_CONF_OK)
-          logger_->log_error("PublishKafka: configure delivery guarantee error result [%s]", errstr);
-      }
-      value = "";
-      if (context->getProperty(RequestTimeOut, value, flowFile) && !value.empty()) {
-        core::TimeUnit unit;
-        if (core::Property::StringToTime(value, valInt, unit) && core::Property::ConvertTimeUnitToMS(valInt, unit, valInt)) {
-          valueConf = std::to_string(valInt);
-          rd_kafka_topic_conf_set(topic_conf_, "request.timeout.ms", valueConf.c_str(), errstr, sizeof(errstr));
-          logger_->log_debug("PublishKafka: request.timeout.ms [%s]", valueConf);
-          if (result != RD_KAFKA_CONF_OK)
-            logger_->log_error("PublishKafka: configure timeout error result [%s]", errstr);
-        }
-      }
+  // Get some properties not (only) used directly to set up librdkafka
+  std::string value;
+
+  // Batch Size
+  uint32_t batch_size;
+  value = "";
+  if (context->getProperty(BatchSize.getName(), value) && !value.empty() && core::Property::StringToInt(value, batch_size)) {
+    logger_->log_debug("PublishKafka: Batch Size [%lu]", batch_size);
+  } else {
+    batch_size = 10;
+  }
 
-      std::shared_ptr<KafkaTopic> kafkaTopicref = std::make_shared<KafkaTopic>(topic_reference, topic_conf_);
+  // Target Batch Payload Size
+  uint64_t target_batch_payload_size;
+  value = "";
+  if (context->getProperty(TargetBatchPayloadSize.getName(), value) && !value.empty() && core::Property::StringToInt(value, target_batch_payload_size)) {
+    logger_->log_debug("PublishKafka: Target Batch Payload Size [%llu]", target_batch_payload_size);
+  } else {
+    target_batch_payload_size = 512 * 1024U;
+  }
 
-      conn->putTopic(topic, kafkaTopicref);
-    }
+  // Max Flow Segment Size
+  uint64_t max_flow_seg_size;
+  value = "";
+  if (context->getProperty(MaxFlowSegSize.getName(), value) && !value.empty() && core::Property::StringToInt(value, max_flow_seg_size)) {
```

Comment:
There is a function to do the check and conversion for you:
```
  template<typename T>
  bool getProperty(const std::string &name, T &value) const {
    return getPropertyImp<typename std::common_type<T>::type>(name, value);
  }
```

It's also fine in scope of a follow-up.

Reply:
I agree it would be better, but in this Processor this pattern is used, and I don't want to change everything now. Will create a follow-up.